userspace/libsinsp/CMakeLists.txt

```diff
@@ -203,8 +203,13 @@ set(SINSP_LIBRARIES
 	"${CARES_LIB}"
 	"${RE2_LIB}")
 
-add_dependencies(sinsp valijson)
-add_dependencies(sinsp re2)
+if(USE_BUNDLED_VALIJSON)
+	add_dependencies(sinsp valijson)
+endif()
+
+if(USE_BUNDLED_RE2)
+	add_dependencies(sinsp re2)
```

Comment:
Thanks for the fix!
I don't get how this was working when using system deps :/ i mean, using system deps the `valijson` and `re2` target do not exist! But both CI (and my local builds) work fine.
Perhaps it behaves differently on old cmake versions?
@jasondellaluce

Reply:
Fort the CI we have https://github.com/falcosecurity/libs/blob/master/.github/install-deps.sh. For your local build, I can't say! I guess some Arch Linux distros bundle stuff like re2.

Reply:
No the problem is that we were adding a dependency on an unexisting target; that should be an error!
But again, perhaps it depends on the cmake version, and newer cmake is less grumpy about that?

Reply:
It does depend on the CMake version. According to the [CMP0046](https://cmake.org/cmake/help/latest/policy/CMP0046.html) page, older CMake versions ignore the dependency, while versions 3.0 and later treat it as an error. We set `cmake_minimum_required(VERSION 2.8.5)` in the top-level CMakeLists.txt, so the old behavior is used. If I change that to 3.0.0 or add

```
if(POLICY CMP0046)
        cmake_policy(SET CMP0046 NEW)
endif()
```

I get an error without the `USE_BUNDLED_RE2` conditional.

Reply:
Thanks for the clarification Gerald! Makes sense!